test/e2e/knative_serving_test.go

```diff
@@ -73,16 +75,19 @@ func TestKnativeServing(t *testing.T) {
 		testKnativeVersusKubeServicesInOneNamespace(t, caCtx)
 	})
 
-	// t.Run for proxy testing
-	t.Skip("update global proxy and verify calls goes through proxy server")/*
-		func(t *testing.T) {
-				testKnativeServingForGlobalProxy(t, caCtx)
-			}
-	*/
+	t.Run("update global proxy and verify calls goes through proxy server", func(t *testing.T) {
+		testKnativeServingForGlobalProxy(t, caCtx)
+	})
 
 	t.Run("remove knativeserving cr", func(t *testing.T) {
-		if err := v1a1test.DeleteKnativeServing(caCtx, knativeServing, knativeServing); err != nil {
-			t.Fatal("Failed to remove Knative Serving", err)
+		err := v1a1test.DeleteKnativeServing(caCtx, knativeServing, knativeServing)
+		if err != nil {
+			// Sometimes because of proxy unset takes time to login so need to wait until pods are up again
+			if strings.Contains(err.Error(), "Unauthorized") {
+				test.WaitForControllerEnvironment(caCtx, knativeServing, proxyVerificationEnvName)
```

Comment:
Let's put this in the test above. The test updating the proxy should make sure that once it finishes, the state is as before, i.e.

1. Proxy has been unset
2. Everything has updated accordingly

We shouldn't bleed that over into the next test as that's going to cause maintainability issues down the line.